src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtAddressServiceTest.java

```diff
@@ -0,0 +1,153 @@
+            new uk.gov.hmcts.dts.fact.entity.CourtAddress(MOCK_COURT, VISIT_OR_CONTACT_US_ADDRESS_TYPE, emptyList(), emptyList(), null, null, null),
+            new uk.gov.hmcts.dts.fact.entity.CourtAddress(MOCK_COURT, WRITE_TO_US_ADDRESS_TYPE, emptyList(), emptyList(), null, null, null),
+            new uk.gov.hmcts.dts.fact.entity.CourtAddress(MOCK_COURT, WRITE_TO_US_ADDRESS_TYPE, emptyList(), emptyList(), null, null, null),
+            new uk.gov.hmcts.dts.fact.entity.CourtAddress(MOCK_COURT, VISIT_US_ADDRESS_TYPE, emptyList(), emptyList(), null, null, null)
+        );
+        when(MOCK_COURT.getAddresses()).thenReturn(courtAddresses);
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(MOCK_COURT));
+
+        final List<CourtAddress> results = adminCourtAddressService.getCourtAddressesBySlug(COURT_SLUG);
+        assertThat(results).hasSize(courtAddresses.size());
+        assertThat(results.subList(0, 3).stream().map(CourtAddress::getAddressTypeId))
+            .allMatch(c -> c.equals(VISIT_US_ADDRESS_TYPE_ID) || c.equals(VISIT_OR_CONTACT_US_ADDRESS_TYPE_ID));
+        assertThat(results.subList(3, 6).stream().map(CourtAddress::getAddressTypeId))
+            .allMatch(c -> c.equals(WRITE_TO_US_ADDRESS_TYPE_ID));
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenRetrievingAddressesForNonExistentCourt() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> adminCourtAddressService.getCourtAddressesBySlug(COURT_SLUG))
+            .isInstanceOf(NotFoundException.class)
+            .hasMessage(NOT_FOUND + COURT_SLUG);
+    }
+
+    @Test
+    void shouldUpdateCourtAddresses() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.of(MOCK_COURT));
+        when(adminAddressTypeService.getAddressTypeMap()).thenReturn(Map.of(VISIT_US_ADDRESS_TYPE_ID, VISIT_US_ADDRESS_TYPE,
+                                                                            WRITE_TO_US_ADDRESS_TYPE_ID, WRITE_TO_US_ADDRESS_TYPE));
+        when(courtAddressRepository.saveAll(any())).thenReturn(COURT_ADDRESSES_ENTITY);
+
+        final List<CourtAddress> results = adminCourtAddressService.updateCourtAddresses(COURT_SLUG, EXPECTED_ADDRESSES);
+        assertThat(results).hasSize(ADDRESS_COUNT);
+        assertThat(results.get(0)).isEqualTo(VISIT_US_ADDRESS);
+        assertThat(results.get(1)).isEqualTo(WRITE_TO_US_ADDRESS);
+
+        verify(courtAddressRepository).deleteAll(any());
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenUpdatingAddressesForNonExistentCourt() {
+        when(courtRepository.findBySlug(COURT_SLUG)).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> adminCourtAddressService.updateCourtAddresses(COURT_SLUG, any()))
+            .isInstanceOf(NotFoundException.class)
+            .hasMessage(NOT_FOUND + COURT_SLUG);
```

Comment:
Super minor again, string format if you want. Think the reason is because of readability and/or performance/bug reduction - up to you though!